modules/core/src/commonMain/kotlin/fr/acinq/lightning/json/JsonSerializers.kt

```diff
@@ -286,16 +289,24 @@ object JsonSerializers {
     object SignedSharedTransactionSerializer
 
     @Serializable
-    data class InteractiveTxSigningSessionSurrogate(val fundingParams: InteractiveTxParams, val fundingTxId: TxId)
+    data class InteractiveTxSigningSessionSurrogate(val fundingParams: InteractiveTxParams, val fundingTxId: TxId, val localCommit: Either<UnsignedLocalCommit, LocalCommit>, val remoteCommit: RemoteCommit)
     object InteractiveTxSigningSessionSerializer : SurrogateSerializer<InteractiveTxSigningSession, InteractiveTxSigningSessionSurrogate>(
-        transform = { s -> InteractiveTxSigningSessionSurrogate(s.fundingParams, s.fundingTx.txId) },
+        transform = { s -> InteractiveTxSigningSessionSurrogate(s.fundingParams, s.fundingTx.txId, s.localCommit, s.remoteCommit) },
         delegateSerializer = InteractiveTxSigningSessionSurrogate.serializer()
     )
 
     @Serializer(forClass = RbfStatus::class)
     object RbfStatusSerializer
 
-    object SpliceStatusSerializer : StringSerializer<SpliceStatus>({ it::class.simpleName!! })
+    @Serializer(forClass = SpliceStatus.WaitingForSigs::class)
+    object SpliceStatusWaitingForSigsSerializer
+
+    @Serializable
+    data class SpliceStatusSurrogate(val status: String, val waitingForSigs: SpliceStatus.WaitingForSigs? = null)
+    object SpliceStatusSerializer : SurrogateSerializer<SpliceStatus, SpliceStatusSurrogate>(
+        transform = { s -> SpliceStatusSurrogate(s::class.simpleName!!, s as? SpliceStatus.WaitingForSigs) },
+        delegateSerializer = SpliceStatusSurrogate.serializer()
+    )
```

Comment:
Unfortunately, had to update the json serializers, otherwise the golden record test was not doing much.